src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -119,6 +119,23 @@ namespace
         {
             return pal::pal_utf8string(get_filename_without_ext(context->application), value_buffer, value_buffer_size);
         }
+        else if (::strcmp(key, HOST_PROPERTY_COMMAND_LINE) == 0)
+        {
+            pal::string_t command_line = context->host_path;
+            if (context->host_mode == host_mode_t::muxer)
+            {
+                command_line.append(_X(" "));
+                command_line.append(context->application);
+            }
+
+            for (int i = 0; i < context->app_argc; ++i)
+            {
+                command_line.append(_X(" "));
+                command_line.append(context->app_argv[i]);
+            }
+
+            return pal::pal_utf8string(command_line, value_buffer, value_buffer_size);
+        }
```

Comment:
Not necessarily for this PR, but I think we should figure out the design of the "on-demand" properties and their interaction with `hostfxr_get_runtime_property_value` / `hostfxr_set_runtime_property_value`.

My immediate reaction would be: Calling `hostfxr_set_runtime_property_value("COMMAND_LINE", ...)` should work and the specified value should be seen by the runtime.
There's a separate discussion about the `COMMAND_LINE` property and its interaction with `Environment.GetCommandLineArgs` as well.